static/app/utils/useTeams.tsx

```diff
@@ -0,0 +1,185 @@
+import {useState} from 'react';
+import uniqBy from 'lodash/uniqBy';
+
+import TeamActions from 'app/actions/teamActions';
+import {Client} from 'app/api';
+import OrganizationStore from 'app/stores/organizationStore';
+import TeamStore from 'app/stores/teamStore';
+import {useLegacyStore} from 'app/stores/useLegacyStore';
+import {Team} from 'app/types';
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+import RequestError from 'app/utils/requestError/requestError';
+import useApi from 'app/utils/useApi';
+
+type State = {
+  /**
+   * This is state for when fetching data from API
+   */
+  fetching: boolean;
+  /**
+   * The error that occurred if fetching failed
+   */
+  fetchError: null | RequestError;
+  /**
+   * Indicates that Team results (from API) are paginated and there are more
+   * Teams that are not in the initial response.
+   */
+  hasMore: null | boolean;
+  /**
+   * The last query we searched. Used to validate the cursor
+   */
+  lastSearch: null | string;
+  /**
+   * Pagination
+   */
+  nextCursor?: null | string;
+};
+
+export type Result = {
+  /**
+   * The loaded teams list
+   */
+  teams: Team[];
+  /**
+   * This is an action provided to consumers for them to update the current
+   * teams result set using a simple search query. You can allow the new
```

Comment:
this comment can be changed to remove the bit about appending/replacing

Reply:
Ah yea